src/lib/operators/table_scan.cpp

```diff
@@ -0,0 +1,198 @@
+        ValueID size = static_cast<ValueID>(attribute_vector->size());
+
+        switch (scan_type()) {
+          case ScanType::OpEquals: {
+            auto value = typed_dict_segment->find_in_dict(search_value());
+            for (auto index = ValueID(0); index < size; index++) {
+              auto attribute = typed_dict_segment->attribute_vector()->get(i);
+              if (attribute == value) {
+                posList.emplace_back(RowID({i, ChunkOffset(index)}));
+              }
+            }
+            break;
+          }
+          case ScanType::OpNotEquals: {
+            auto value = typed_dict_segment->find_in_dict(search_value());
+            for (auto index = ValueID(0); index < size; index++) {
+              auto attribute = typed_dict_segment->attribute_vector()->get(i);
+              if (attribute != value) {
+                posList.emplace_back(RowID({i, ChunkOffset(index)}));
+              }
+            }
+            break;
+          }
+          case ScanType::OpLessThan: {
+            auto value = typed_dict_segment->find_in_dict(search_value());
+            for (auto index = ValueID(0); index < size; index++) {
+              auto attribute = typed_dict_segment->attribute_vector()->get(i);
+              if (attribute < value) {
+                posList.emplace_back(RowID({i, ChunkOffset(index)}));
+              }
+            }
+            break;
+          }
+          case ScanType::OpLessThanEquals: {
+            auto value = typed_dict_segment->find_in_dict(search_value());
+            for (auto index = ValueID(0); index < size; index++) {
+              auto attribute = typed_dict_segment->attribute_vector()->get(i);
+              if (attribute <= value) {
+                posList.emplace_back(RowID({i, ChunkOffset(index)}));
+              }
+            }
+            break;
+          }
+          case ScanType::OpGreaterThan: {
+            auto value = typed_dict_segment->find_in_dict(search_value());
+            for (auto index = ValueID(0); index < size; index++) {
+              auto attribute = typed_dict_segment->attribute_vector()->get(i);
+              if (attribute > value) {
+                posList.emplace_back(RowID({i, ChunkOffset(index)}));
+              }
+            }
+            break;
+          }
+          case ScanType::OpGreaterThanEquals: {
+            auto value = typed_dict_segment->find_in_dict(search_value());
+            for (auto index = ValueID(0); index < size; index++) {
+              auto attribute = typed_dict_segment->attribute_vector()->get(i);
+              if (attribute >= value) {
+                posList.emplace_back(RowID({i, ChunkOffset(index)}));
+              }
+            }
+            break;
+          }
+        }
+
+      const auto typed_ref_segment = std::dynamic_pointer_cast<ReferenceSegment>(segment);
+      if(typed_ref_segment != nullptr) {
+        auto pos_list_size = typed_ref_segment->pos_list()->size();
```

Comment:
Sometimes you get the size before the for loop, sometimes you use `size()` in the for loop. This should be unified